internal/config/settings.go

```diff
@@ -155,6 +155,68 @@ func init() {
 	VolatileSettings = make(map[string]bool)
 }
 
+func validateParsedSettings() error {
+	var err error
+	defaults := DefaultAllSettings()
+	for k, v := range parsedSettings {
+		if strings.HasPrefix(reflect.TypeOf(v).String(), "map") {
+			if strings.HasPrefix(k, "ft:") {
+				for k1, v1 := range v.(map[string]interface{}) {
+					if _, ok := defaults[k1]; ok && !verifySetting(k1, reflect.TypeOf(v1), reflect.TypeOf(defaults[k1])) {
+						err = fmt.Errorf("Error: setting '%s' has incorrect type (%s), using default value: %v (%s)", k1, reflect.TypeOf(v1), defaults[k1], reflect.TypeOf(defaults[k1]))
+						parsedSettings[k].(map[string]interface{})[k1] = defaults[k1]
+						continue
+					}
+					if e := OptionIsValid(k1, v1); e != nil {
```

Comment:
Maybe do this inside `verifySetting()`? That would slim `validateParsedSettings()` down quite a bit.